R/system.R

```diff
@@ -2445,6 +2470,34 @@ get_excel_sheets <- function(path){
   }
 }
 
+#'API that imports multiple same structure CSV files and merge it to a single data frame
+#'@export
+read_delim_files <- function(files, delim, quote = '"',
+                              escape_backslash = FALSE, escape_double = TRUE,
+                              col_names = TRUE, col_types = NULL,
+                              locale = readr::default_locale(),
+                              na = c("", "NA"), quoted_na = TRUE,
+                              comment = "", trim_ws = FALSE,
+                              skip = 0, n_max = Inf, guess_max = min(1000, n_max),
+                              progress = interactive(), with_api_key = FALSE) {
+    # set name to the files so that it can be used for the "id" column created by purrr:map_dfr.
+    files <- setNames(as.list(files), files)
+    df <- purrr::map_dfr(files, exploratory::read_delim_file,delim = delim, quote = quote,
+                   escape_backslash = escape_backslash, escape_double = escape_double,
+                   col_names = col_names, col_types = col_types,
+                   locale = locale,
+                   na = na, quoted_na = quoted_na,
+                   comment = comment, trim_ws = trim_ws,
+                   skip = skip, n_max = n_max, guess_max = guess_max,
+                   progress = progress, with_api_key = with_api_key, .id = "exp.file.id") %>% mutate(exp.file.id = basename(exp.file.id))  # extract file name from full path with basename and create file.id column.
+    id_col <- avoid_conflict(colnames(df), "id")
+    # copy internal exp.file.id to the id column.
+    df[[id_col]] <- df[["exp.file.id"]]
+    # drop internal column and move the id column to the very beginning.
+    df %>% dplyr::select(!!!rlang::syms(id_col), dplyr::everything(), -exp.file.id)
```

Comment:
!!rlang::sym(id_col) might be more appropriate.

Reply:
Changed as suggested.